### ISSUE TITLE
Speech stops/cuts mid-sentence with GTTSEngine and EdgeEngine

### ISSUE THREAD

**Issue by ArsenicBismuth** [KoljaB/RealtimeTTS#223]
Hi, so I wanted to try some free TTS API, which is why I'm here now. But for some reason the TTS would always cut midway of the input, in the strangest place.

Example, with this input:
`Oh! Let me tell you about something that happened to me at exactly at 12pm on November 29th, 2024! I was in professor's robotics lab, working late on a project.`

It'd say something like this (`--` to indicate long pause):
`Oh! Let me tell you about something that happened to me at exactly at -- 12pm on November 29th, 2024! -- I was in pro--fessor's robotics lab, working late on a project.`

I'm using `play_async()`, but similar behavior appear when using `play()` (tho, not as bad).

This behavior is apparent on those two engines I'm able to test.

**Comment by ArsenicBismuth** [comment]
Is there maybe something about sentence length limit that I'm unaware of? So for example I should've cut it into multiple valid sentences, so that the TTS won't automatically cut it into improper ones.

**Comment by KoljaB** [comment]
Will look into that, thank you for pointing this out.

**Comment by ArsenicBismuth** [comment]
@KoljaB So I tested again with `SystemEngine`, and I got similar random cuts. And the cuts often sometimes cut mid-word, like "excite--d", "trad--ing". As if I'm streaming incomplete word (I'm not, I gave it full prompt result without streaming).

For reference, before using your package, I tested using the engines in other python module, such as SystemEngine (`PyTTSx3`) & EdgeEngine (`edge_tts`), neither has this behavior.

**Comment by ArsenicBismuth** [comment]
I setup an Azure account to check `AzureEngine`, and turns out this engine has no such issue. It speaks flawlessly to the end with no strange cut at all.

**Comment by KoljaB** [comment]
It might have to do with [this](https://github.com/KoljaB/RealtimeTTS/commit/41a68561c7878ceb48c2384f988ff3a71cace274) commit. 

Do you have the same problems when using version v0.4.10?
```bash
pip install realtimetts[all]==0.4.10
```

**Comment by ArsenicBismuth** [comment]
@KoljaB No, the way it "cuts midway" is NOT like the audio is cut abruptly (like when you pause a video). You can check below the audio (sorry for not doing that earlier):

https://github.com/user-attachments/assets/7b47ec34-deda-4f84-a402-d197c079ac3d

Here you can hear the "check" is split into "chec" and "k" with its proper "kay" alphabet sound. So, it's more like the text input are split into chunks. Ofc, you can also hear the long pauses after certain words.

**Comment by KoljaB** [comment]
You're right, this has clearly nothing to do with audio streaming. 
Can I see the parameters for the calls to TextToAudioStream constructor (tokenizer, language etc) and play method (lots of parameters here have influence on the sentence splitting)?

**Comment by ArsenicBismuth** [comment]
This is the code, nothing else:

```python
class SpeechEngine:

    def __init__(self, voice: int):

        self.voice_engine = EdgeEngine()
        self.voice_engine.set_voice(voice)
        self.voice_stream = TextToAudioStream(self.voice_engine, language="en")

    def speak_text_async(self, text: str):
        self.voice_stream.feed(text).play_async()
```

**Comment by KoljaB** [comment]
Ahh, now I can reproduce. There's something deeply wrong in the sentence splitting.  

Still unsure what, but now I will find it. Thank you so much for providing all the information, that was great help!

**Comment by KoljaB** [comment]
Should be solved in [v0.4.14](https://github.com/KoljaB/RealtimeTTS/releases/tag/v0.4.14) now. Thanks again for best support possible!

**Comment by ArsenicBismuth** [comment]
Ok yeah this fixed it! Thank you. It's now pausing properly at the end of sentences.
